Sprint3/step 1

i0ji opened the pull request:
First attempt of the third sprint task:
Routing, auth, register, login and out, logic, multiply edits, refactored ingredient modal;
Regactored app/main markup according to the task and specificbof using routing;

Commit:
build

i0ji committed:
19.02.24/b:sprint3/step-1/v:0.1.7.3
react-router lib added; routing started for "constructor" and "404" pages;

i0ji committed:
Components transition; Router added;Login page in progress;

i0ji committed:
Merge remote-tracking branch 'origin/sprint3/step-1' into sprint3/step-1

# Conflicts:
#	.yarn/install-state.gz

Commit:
style: minor CSS changes in construcotr list;

Commit:
style: base markup refactor;
feat: login page set

Commit:
v:0.1.8.0
style: base markup refactor;
feat: login/forgot/register pages set; w/o logic

i0ji committed:
20.02.24/b:sprint3/step-1/v:0.1.8.0.1
style:multiply edits and reworks, notFoundPage;

i0ji committed:
feature: routing

i0ji committed:
refactor: move Router according to the task;
style: get rid of endless style namings

i0ji committed:
21.02.24/b:sprint3/step-1/v:0.1.8.1
style: styling active NavLinks in header

i0ji committed:
feature: modal+route

Commit:
v:0.1.8.0.2
style: small css edits;
feature: wrapping ingredient card with link, setting styles;

i0ji committed:
22.02.24/b:sprint3/step-1/v:0.1.8.2
feature: modal for ingredients;
styles: multiply scss edits;
refactor: styling of active/inactive header buttons

i0ji committed:
/tag

Commit:
23.02.24/b:sprint3/step-1/v:0.1.8.0.3

refactoring modals and Ingredient pages;

Commit:
v:0.1.8.0.2
style: small css edits;
feature: setting modal and separate window of ingredient available via direct url;

Commit:
chore

Commit:
style: animated mount and unmount;
style: background of direct ingredeint url;

Commit:
attempts of making animation

Commit:
attempts of making animation

Commit:
attempts of making animation

Commit:
25.02.24/b:sprint3/step-1/v:0.1.8.1
feature: login/auth/get/protected route/tokens;
styles: loading refactored;

Commit:
feature: reset/register/get tokens/utils

Commit:
feature: reset/register/get tokens/utils

Commit:
feature: reset/register/get tokens/utils

i0ji committed:
Merge branch 'sprint3/step-1' of https://github.com/i0ji/p_stellar_burger into sprint3/step-1

# Conflicts:
#	src/pages/IngredientDetailsPage/IngredientDetailsPage.tsx

i0ji committed:
26.02.24/b:sprint3/step1/v:0.1.8.2
get two tokens;auth logic;custom hook for forms

Commit:
feature:ProtectedRoute

i0ji committed:
27.02.24/b:sprint3/step-1/v:0.1.8.3
feature: auth, protected, statement, login and logout logic, useForm

i0ji committed:
Merge remote-tracking branch 'origin/sprint3/step-1' into sprint3/step-1

# Conflicts:
#	src/App.tsx
#	src/components/common/ProtectedRoute/ProtectedRoute.tsx
#	src/pages/LoginPage.tsx
#	src/services/slices/authSlice.ts

Commit:
28.02.24/b:sprint3/step-1/v:0.1.8.4
feature: setAuthCheck rework; routing rework;
types declaration;

Commit:
28.02.24/b:sprint3/step-1/v:0.1.8.4
feature: handleRegister; ProtectedRoute logic

Commit:
register checked

Commit:
register checked

andryxxa93 commented:
Здравствуйте! Проделана большая работа 👍 

Сильные стороны работы:
- Есть ProtectedRoute
- Реализован роутинг модальных окон
- Верно определяется активная ссылка в хедере

Однако есть пару моментов, которые необходимо исправить, чтобы работа была принята.

Необходимо исправить:
1. После неудачного запроса на регистрацию всё равно кидает на страницу профиля.

2. Авторизация сбрасывается при перезагрузке страницы

3. После отправки запроса на password-reset необходимо переводить юзера на страницу /reset-password, где надо будет ввести новый имейл и код из письма. Сейчас его кидает на страницу логина.

4. Не авторизованный юзер не должен иметь возможность сделать заказ

5.  на тег form нужно навешивать обработчик onSubmit, а не обработчик onClick на кнопку сабмита.

6. На странице регистрации нельзя сделать видимым введённый пароль

7. На странице профиля в инпутах после регистрации не отображаются данные юзера. Не хватает инпута с имейлом. Так же в них не получается ничего ввести. При их изменении должны появляться кнопки "Отмена" и "Сохранить"

8. Роута logout нет в задании. При нажатии на "Выход" необходимо делать запрос на ендпоинт logout. Пожалуйста обратите внимание на этот запрос в тех-задании

9. Функция fetchWithRefresh объявлена, но нигде не используется

**Напомню, чтобы работа была принята нужно исправить все замечания помеченные как **необходимо исправить**.

Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка F12) перед отправкой на ревью. 

Удачных правок!

Commit:
02.03.24/b:sprint3/step-1/v:0.1.8.5;
onCLick changed to logically correct, password hide/show

Commit:
02.03.24/v:0.1.8.5.1
feature:added wight route from login to '/' after enter;
feature:added password hide/show logic;
in progress: auth/login/logout/reset logic

Commit:
reset password logic and routing after change request

Commit:
10 sec redirection and countdown after successfully changing password;

Commit:
feature: prevent unAuth oreder logic;

Commit:
03.03.24/b:sprint3/step-1/v:0.1.8.5
feature:logout logic,
refactor: replaced auth/login/logout logic back to utils API;

Commit:
v:0.1.8.5
refactoring profilePage

Commit:
v:0.1.8.5
feature: sending updated data works partly

Commit:
v:0.1.8.5
feature: sending updated data works partly

i0ji committed:
04.03.24b:sprint3/step-1/v:0.1.8.5
global: yarn updated
fix: logic and display of the component of Profile;
fix: "save/cancel" buttons
style: appearance of icons on editing
chore: some minor edits in api.ts

i0ji committed:
yarn release file

i0ji committed:
05.03.24/b:sprint3/step1/v:0.1.8.5
feature: save auth

i0ji committed:
05.03.24/b:sprint3/step1/v:0.1.8.6
style: one more smooth animation of buttons and modal

Commit:
v:0.1.9.0
feature: use fetchWithRefresh

Commit:
v:0.1.9.0
style: some minor edits

Commit:
v:0.1.9.1
fix: typing store/slice/components

Commit:
v:0.1.9.1
fix: typing store/slice/components
fix:  slow userData update after changing users - added removal function to logout reducer;

Commit:
v:0.1.9.1
fix: typing store/slice/components

Commit:
v:0.1.9.1
fix: typing store/slice/components

andryxxa93 commented:
**Необходимо исправить**
- Из страницы профиля необходимо убрать запрос данных юзера, не то каждый заход на эту страницу вызывает запрос, что не очень хорошо для производительности
- В инпуты на странице профиля невозможно ничего ввести
- Роута logout нет в задании. При нажатии на "Выход" необходимо делать запрос на ендпоинт logout. Пожалуйста обратите внимание на этот запрос в тех-задании

Commit:
06.03.24/b:sprint3/step-1/v:0.1.9.2
fix: logout logic after second review;
fix: fetching userData from app - not profilePage.
refactor: edit slice due to new logout logic

andryxxa93 commented:
Супер, вы отлично поработали!
Поздравляю! Работа принята.
Желаю удачного продолжения обучения!
